### PR TITLE
feat: remove access for service-accounts

### PR DESCRIPTION
**Pull request by mattjoke**
* service accounts are now not able to log in
* a new dialog has been added, that can redirect users to /login or /service-access
* the dialog logic is evaluated when principal is loading

### CODE REVIEW

File: libs/perun/services/src/lib/init-auth.service.ts [hunk 1]
```diff
@@ -109,6 +110,10 @@ export class InitAuthService {
       if (perunPrincipal.user === null) {
         const config = getDefaultDialogConfig();
         this.dialog.open(UserDontExistDialogComponent, config);
+      } else if (perunPrincipal.user.majorSpecificType === 'SERVICE') {
```
Comment (bodnara) [inline libs/perun/services/src/lib/init-auth.service.ts:113]:
```suggestion
      } else if (perunPrincipal.user.serviceUser) {
```

File: libs/perun/services/src/lib/init-auth.service.ts [hunk 2]
```diff
@@ -109,6 +110,10 @@ export class InitAuthService {
...
+        this.authService.logout();
```
Comment (bodnara) [inline libs/perun/services/src/lib/init-auth.service.ts:114]: Depending on the proxy and configuration the logout might redirect you without the dialog ever showing. But even if the logout was performed at later stage the redirect would mess up the flow.

File: libs/general/src/lib/user-not-allowed-access/user-not-allowed-access.component.html [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+</h2>
+
+<mat-dialog-actions align="end">
+  <button mat-flat-button color="primary" (click)="redirectToLogin()">
+    {{'SHARED_LIB.PERUN.COMPONENTS.USER_NOT_ALLOWED.LOGIN' | translate}}
+  </button>
+  <button mat-flat-button color="primary" (click)="redirectToServiceLogin()">
+    {{'SHARED_LIB.PERUN.COMPONENTS.USER_NOT_ALLOWED.SERVICE_LOGIN' | translate}}
+  </button>
+</mat-dialog-actions>
```
Comment (bodnara) [inline libs/general/src/lib/user-not-allowed-access/user-not-allowed-access.component.html:16]: Fixes the dialog, layout and colors.
```suggestion
<h1 mat-dialog-title>{{'SHARED_LIB.PERUN.COMPONENTS.USER_NOT_ALLOWED.TITLE' | translate}}</h1>
<div mat-dialog-content class="dialog-container">
  {{'SHARED_LIB.PERUN.COMPONENTS.USER_NOT_ALLOWED.MESSAGE' | translate}}
</div>
<div mat-dialog-actions>
  <button mat-flat-button class="ms-auto" color="accent" (click)="redirectToLogin()">
    {{'SHARED_LIB.PERUN.COMPONENTS.USER_NOT_ALLOWED.LOGIN' | translate}}
  </button>
  <button mat-flat-button color="accent" (click)="redirectToServiceLogin()">
    {{'SHARED_LIB.PERUN.COMPONENTS.USER_NOT_ALLOWED.SERVICE_LOGIN' | translate}}
  </button>
</div>
```

File: apps/admin-gui/src/assets/i18n/en.json [hunk 4]
```diff
@@ -3104,6 +3104,12 @@
         "USER_DONT_EXIST": {
           "TITLE": "The requested user (by Id or external identity) doesn't exist."
         },
+        "USER_NOT_ALLOWED": {
+          "TITLE": "Access not allowed",
+          "MESSAGE": "Service accounts are not allowed to access this application.",
+          "LOGIN": "Redirect to /login",
+          "SERVICE_LOGIN": "Redirect to /service-access"
```
Comment (bodnara) [inline apps/admin-gui/src/assets/i18n/en.json:3111]: This a maybe a preference thing but this is not very elegant. I would honestly maybe put here "Change account" and redirect to login page only.

**Review by bodnara — CHANGES_REQUESTED** [review]

File: libs/perun/services/src/lib/init-auth.service.ts (hunk 2 above)
Comment (mattjoke) [reply libs/perun/services/src/lib/init-auth.service.ts:114]: We could detect the proxy configuration and then we could do logout with a redirect, i.e. logout would happen after the dialog was closed.

File: apps/admin-gui/src/assets/i18n/en.json (hunk 4 above)
Comment (mattjoke) [reply apps/admin-gui/src/assets/i18n/en.json:3111]: What should be the correct way when someone logs in from the service-access? Should it redirect to it? Or just to the login page?

Comment (bodnara) [reply apps/admin-gui/src/assets/i18n/en.json:3111]: I guess the best option for user is to redirect to login option that was used before. So for OIDC to /login and BA to /service-access.

**Review by HejdaJakub — APPROVED** [review]

**Review by bodnara — APPROVED** [review]

**Pull request merged**